src/main/java/net/sf/marineapi/nmea/event/AbstractSentenceListener.java

```diff
@@ -101,9 +101,15 @@ public void readingStopped() {
 	@SuppressWarnings("unchecked")
 	public void sentenceRead(SentenceEvent event) {
 		Sentence sentence = event.getSentence();
-		Class<?>[] interfaces = sentence.getClass().getInterfaces();
-		if (Arrays.asList(interfaces).contains(this.expectedType)) {
-			sentenceRead((T) sentence);
+		if (this.expectedType instanceof Class<?>) {
```

Comment:
this.expectedType has type Type which could be a TypeVariable, a ParameterizedType, a TypeVariable or a GenericArrayType in addition to Class. That's why I think that generally this instanceof check is required.